### PR TITLE
pipewire: 0.3.42 -> 0.3.43

### PR DESCRIPTION
**Pull request by jansol**
###### Motivation for this change

https://gitlab.freedesktop.org/pipewire/pipewire/-/releases/0.3.43

- pulse-server should use less memory and overall work better
- filter-chain now supports lv2 effects
- libcamera bump since it had an API change that affects spa-libcamera
- Works On My Machine (tm)

NB: backport requires #153124 to land first (feel free to retarget it directly to the release branch, I may have been overly cautious with sending it to staging)

https://git.libcamera.org/libcamera/libcamera.git/log/?id=40f5fddca7f774944a53f58eeaebc4db79c373d8..1db1e31e664c1f613dc964d8519fe75d67b154b6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
LGTM and is needed for latest Wireplumber (see linked PR)

File: pkgs/development/libraries/libcamera/default.nix [hunk 1]
```diff
@@ -65,7 +81,11 @@ stdenv.mkDerivation {
     openssl
   ];
 
-  mesonFlags = [ "-Dv4l2=true" "-Dqcam=disabled" ];
+  mesonFlags = [
+    "-Dv4l2=${mesonBool true}"
```
Comment (jtojnar) [inline pkgs/development/libraries/libcamera/default.nix:85]: Why add functions when the values are just static?

File: pkgs/development/libraries/libcamera/default.nix [hunk 2]
```diff
@@ -44,13 +54,19 @@ stdenv.mkDerivation {
...
+    # hotplugging
+    systemd
 
     # lttng tracing
     lttng-ust
 
     gtest
   ];
 
+  FONTCONFIG_FILE = fontsConf;
```
Comment (jtojnar) [inline pkgs/development/libraries/libcamera/default.nix:68]: Probably should go next to the other env vars. And a comment why it is needed would be useful.

Comment (jansol) [reply pkgs/development/libraries/libcamera/default.nix:68]: I'm not even sure if this is the proper way to get rid of the fontconfig errors, it's just what I found was being done in other packages. It only gets rid of the "no config file" errors anyway, there are still like a thousand lines of "no writable cache".

File: pkgs/development/libraries/libcamera/default.nix (hunk 1 above)
Comment (jansol) [reply pkgs/development/libraries/libcamera/default.nix:85]: Future proofing mostly (for making the features accessible as parameters to the derivation), and to make the types of the options more explicit.

File: pkgs/development/libraries/libcamera/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/development/libraries/libcamera/default.nix:68]: Yeah, this is the proper “fix”. I am mostly interested what needs fonts in libcamera. It is typically needed by GUI tests or sometimes docs generation.

File: pkgs/development/libraries/libcamera/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/libcamera/default.nix:85]: I would not say it does a very good job in either – it adds clutter to the expression that might (or might not) be used in the future and actually obscures the value that is passed to the options. And the type that accepts `enabled`/`disabled` is called `feature`.

Comment (jansol) [reply pkgs/development/libraries/libcamera/default.nix:85]: Fair enough, I'll just remove it for now. And rename the one in the pipewire expression (where I copied this from) to mesonFeature.

File: pkgs/development/libraries/libcamera/default.nix (hunk 2 above)
Comment (jansol) [reply pkgs/development/libraries/libcamera/default.nix:68]: Probably docs yes, unless gstreamer requires it.

Comment (jtojnar) [reply pkgs/development/libraries/libcamera/default.nix:68]: What commands precede the warning?

Comment (jansol) [reply pkgs/development/libraries/libcamera/default.nix:68]: Seems indeed to be to the documentation. Build log says
```
[256/378] Generating doxygen with a custom commandap_ipa_proxyxyrialization_test.cpp.oK[Kerializer_test.cpp.o.o
/build/libcamera-1db1e31/src/libcamera/base/file.cpp:49: warning: no matching class member found for
  libcamera::File::MapFlag::NoOption
<a few more similar warnings but this looks like it's interspersed here from another command running in parallel>
sh: latex: not found
error: Problems running latex. Check your installation or look for typos in _formulas.tex and check _formulas.log!
sh: dvips: not found
error: Problems running dvips. Check your installation!
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
...
```

The documentation is built with sphinx, which does mention LaTeX as one backend (for pdf rendering) right on its landing page. There is no option for the documentation format and the `meson.build` for the documentation specifically requests html output so my best guess is that this is sphinx probing the environment to determine which backends are actually usable.

**Comment by jansol** [comment]
Removed the systemd unit path patches that were backported to master the other day that are part of the upstream release.

**Comment by jansol** [comment]
Renamed the helper to `mesonEnableFeature` to match the wireplumber PR, inlined the fontconfig workaround, removed `patchShebang` invocations for files that no longer exist.

**Review by jtojnar — APPROVED** [review]

**Comment by jansol** [comment]
@Artturin release-21.11 and nixos-unstable have diverged a bit by now so it's probably easiest to just copy the current state of nixos-unstable into the stable branch (minus the change of defaulting to wireplumber instead of media-session). Remember that libcamera also needs to be updated.

**Pull request merged**